src/main/java/com/teamscale/upload/TeamscaleUpload.java

```diff
@@ -59,350 +35,53 @@ public static void main(String[] args) throws Exception {
-		} catch (SSLHandshakeException e) {
-			handleSslConnectionFailure(commandLine, e);
-		} finally {
-			// we must shut down OkHttp as otherwise it will leave threads running and
-			// prevent JVM shutdown
-			client.dispatcher().executorService().shutdownNow();
-			client.connectionPool().evictAll();
-		}
+		Map<String, Set<File>> filesByFormat = resolveAndConvertFiles(commandLine);
+		TeamscaleClient.performUpload(commandLine, filesByFormat);
 	}
 
 	/**
-	 * Converts the reports from the internal binary XCode format to a readable
-	 * report that can be uploaded to Teamscale.
+	 * Resolves the files that should be uploaded to Teamscale and converts them to
+	 * the expected formated if needed (e.g., XCode reports).
 	 */
-	private static void convertXCodeReports(Map<String, Set<File>> formatToFiles) throws ConversionException {
-		if (!formatToFiles.containsKey(XCResultConverter.XCODE_REPORT_FORMAT)) {
-			return;
-		}
-
-		Set<File> xcresultBundles = formatToFiles.remove(XCResultConverter.XCODE_REPORT_FORMAT);
-
-		for (File xcodeReport : xcresultBundles) {
-			if (!XCResultConverter.needsConversion(xcodeReport)) {
-				formatToFiles.computeIfAbsent(XCResultConverter.XCODE_REPORT_FORMAT, x -> new HashSet<>())
-						.add(xcodeReport);
-				continue;
-			}
-
-			File workingDirectory = createTemporaryWorkingDirectory();
-			XCResultConverter converter = new XCResultConverter(workingDirectory);
-
-			// Cleanup hook to ensure temporarily created files and directories are deleted
-			// if the user
-			// terminates the application forcefully (e.g. via Ctrl+C).
-			Thread cleanupHook = new Thread(() -> {
-				try {
-					converter.stopProcesses();
-				} catch (ConversionException e) {
-					LogUtils.warn(e.getMessage());
-				}
-				deleteWorkingDirectory(workingDirectory);
-			});
-
-			try {
-				Runtime.getRuntime().addShutdownHook(cleanupHook);
-
-				converter.convert(xcodeReport).forEach(convertedReport -> {
-					formatToFiles.computeIfAbsent(convertedReport.reportFormat, x -> new HashSet<>())
-							.add(convertedReport.report);
-				});
-			} finally {
-				deleteWorkingDirectory(workingDirectory);
-				Runtime.getRuntime().removeShutdownHook(cleanupHook);
-			}
-		}
-	}
-
-	private static void deleteWorkingDirectory(File workingDirectory) {
-		try {
-			FileUtils.deleteDirectory(workingDirectory);
-		} catch (IOException e) {
-			LogUtils.warn("Unable to delete temporary working directory " + workingDirectory.getAbsolutePath() + ": "
-					+ e.getMessage());
-		}
-	}
-
-	private static File createTemporaryWorkingDirectory() throws ConversionException {
-		try {
-			return Files.createTempDirectory(null).toFile();
-		} catch (IOException e) {
-			throw new ConversionException(
-					"Error occurred when trying to create temporary working directory:" + e.getMessage(), e);
-		}
-	}
-
-	private static void handleSslConnectionFailure(CommandLine commandLine, SSLHandshakeException e) {
```

Comment:
For the reviewer: Same here. Had the feeling that this unnecessarily blows up this class. Moved it to `TeamscaleClient`